src/Mod/Sketcher/Gui/CommandCreateGeo.cpp

```diff
@@ -5280,14 +5283,147 @@ CmdSketcherCreateFillet::CmdSketcherCreateFillet()
+    eType           = ForEdit;
+}
+
+void CmdSketcherCreatePointFillet::activated(int iMsg)
+{
+    Q_UNUSED(iMsg);
+    ActivateHandler(getActiveGuiDocument(), new DrawSketchHandlerFillet(1));
+}
+
+bool CmdSketcherCreatePointFillet::isActive(void)
+{
+    return isCreateGeoActive(getActiveGuiDocument());
+}
+
+/// @brief Macro that declares a new sketcher command class 'CmdSketcherCompCreateFillets'
+DEF_STD_CMD_ACLU(CmdSketcherCompCreateFillets)
+
+/**
+ * @brief ctor
+ */
+CmdSketcherCompCreateFillets::CmdSketcherCompCreateFillets()
+  : Command("Sketcher_CompCreateFillets")
+{
+    sAppModule      = "Sketcher";
+    sGroup          = QT_TR_NOOP("Sketcher");
+    sMenuText       = QT_TR_NOOP("Fillets");
+    sToolTipText    = QT_TR_NOOP("Joins lines with a smooth arc");
+    sWhatsThis      = "Sketcher_CompCreateFillets";
+    sStatusTip      = sToolTipText;
+    eType           = ForEdit;
+}
+
+/**
+ * @brief Instantiates the fillet handler when the fillet command activated
+ * @param int iMsg
+ */
+void CmdSketcherCompCreateFillets::activated(int iMsg)
+{
+    if (iMsg == 0) {
+        ActivateHandler(getActiveGuiDocument(), new DrawSketchHandlerFillet(0));
+    } else if (iMsg == 1) {
+        ActivateHandler(getActiveGuiDocument(), new DrawSketchHandlerFillet(1));
+    } else {
+        return;
+    }
+
+    // Since the default icon is reset when enabling/disabling the command we have
+    // to explicitly set the icon of the used command.
+    Gui::ActionGroup* pcAction = qobject_cast<Gui::ActionGroup*>(_pcAction);
+    QList<QAction*> a = pcAction->actions();
+
+    assert(iMsg < a.size());
+    pcAction->setIcon(a[iMsg]->icon());
+}
+
+Gui::Action * CmdSketcherCompCreateFillets::createAction(void)
+{
+    Gui::ActionGroup* pcAction = new Gui::ActionGroup(this, Gui::getMainWindow());
+    pcAction->setDropDownMenu(true);
+    applyCommandData(this->className(), pcAction);
+
+    QAction* oldFillet = pcAction->addAction(QString());
+    oldFillet->setIcon(Gui::BitmapFactory().iconFromTheme("Sketcher_CreateFillet"));
+
+    QAction* pointFillet = pcAction->addAction(QString());
+    pointFillet->setIcon(Gui::BitmapFactory().iconFromTheme("Sketcher_CreateFillet"));
+
+    _pcAction = pcAction;
+    languageChange();
+
+    pcAction->setIcon(Gui::BitmapFactory().iconFromTheme("Sketcher_CreateFillet"));
+    int defaultId = 0;
+    pcAction->setProperty("defaultAction", QVariant(defaultId));
+
+    return pcAction;
+}
+
+void CmdSketcherCompCreateFillets::updateAction(int mode)
+{
+    Gui::ActionGroup* pcAction = qobject_cast<Gui::ActionGroup*>(getAction());
+    if (!pcAction)
+        return;
+
+    QList<QAction*> a = pcAction->actions();
+    int index = pcAction->property("defaultAction").toInt();
+    switch (mode) {
+    case Normal:
+    case Construction: // In case we ever decide we need construction-themed fillet icons
+        a[0]->setIcon(Gui::BitmapFactory().iconFromTheme("Sketcher_CreateFillet"));

```

Comment:
I am not sure there is a use case for a fillet tool that obeys the toggle construction state. Usually this applies to geometry addition, but not to transformation. This is better that we discuss it in the forum thread, to gather opinion from the CAD experts. Please reply there with your impressions.

The rationale is that, if there is no use case, I am very reluctant to include the logic to obey this toggle status, as it would be misleading for somebody reading the code.

Reply:
Sounds like chrisb concurs, as do it.  Easy enough to add it back later if we discover a need.